src/components/Table/utils/TableState.ts

```diff
@@ -375,15 +376,17 @@ export class TableState {
   }
 
   private getMatchedChildrenStates(children: GridDataRow<any>[], map: Map<string, SelectedState>): SelectedState[] {
-    return children
+    const respectedChildren = children.flatMap(getChildrenForDerivingSelectState);
```

Comment:
Question: Before this was just doing `children.map` ... but now we're doing `children.flatMap(something that gets children)`, so are we now effectively processing the grandchildren? 

Oh, I guess `getChildrenForDerivingSelectState` returns both the row itself and its children recursively...

Looks like it doesn't matter that we're now processing children + grandchildren vs. before was only doing children?

Reply:
Right, getChildrenForDerivingSelectState returns both the row itself and its children _only_ if the row has `inferSelectedState === false`. Otherwise, we only return the row. So basically peering deeper into the nesting only if necessary.

Reply:
Cool, and just checking, this line 378 was previously _not_ getting deep results, but it is now, and that's fine / no big deal? Probably not if things are passing tests/etc., but just checking.

Reply:
Correct. It was _not_ getting deep results previously. But now it does get deep results _only_ if necessary.